source/Music.cpp

```diff
@@ -0,0 +1,115 @@
+#include <iostream>
+#include <thread>
+#include "Music.hpp"
+#include "AudioError.hpp"
+
+Music::Music(Musics m, float loop)
+  : source(AL_NONE), loopTime(loop)
+{
+  if (ov_fopen(Audio::getMusicFileName(m), &oggStream) < 0)
+    throw AudioError("Could not open .ogg file");
+  vorbisInfo = ov_info(&oggStream, -1);
+  vorbisComment = ov_comment(&oggStream, -1);
+  format = vorbisInfo->channels == 1 ? AL_FORMAT_MONO16 : AL_FORMAT_STEREO16;
+  alGenSources(1, &source);
+  Audio::checkError();
+  alGenBuffers(2, buffers.data());
+  if (!Audio::checkError(false))
+    {
+      alDeleteSources(1, &source);
+      ov_clear(&oggStream);
+      throw AudioError("Couldn't generate a buffer in Music::Music");
+    }
+
+  alSource3f(source, AL_POSITION, 0., 0., 0.);
+  alSource3f(source, AL_VELOCITY, 0., 0., 0.);
+  alSource3f(source, AL_DIRECTION, 0., 0., 0.);
+  alSourcef(source, AL_ROLLOFF_FACTOR, 0.);
+  alSourcef(source, AL_SOURCE_RELATIVE, AL_TRUE);
+  if (!Audio::checkError(false))
+    {
+      alDeleteSources(1, &source);
+      ov_clear(&oggStream);
+      alDeleteBuffers(1, &buffers[0]);
+      throw AudioError("Couldn't generate a buffer in Music::Music");
+    }
+}
+
+Music::~Music()
+{
+  alSourceStop(source);
+  unqueuePending();
+  alDeleteSources(1, &source);
+  alDeleteBuffers(1, &buffers[0]);
+  ov_clear(&oggStream);
+}
+
+void Music::play(void)
+{
+  if (!streamFile(buffers[0]) || !streamFile(buffers[1]))
+    throw AudioError("Couldn't play music");
+  alSourceQueueBuffers(source, 2, buffers.data());
+  alSourcePlay(source);
+}
+
+void Music::setLoopTime(float f)
+{
+  loopTime = f;
+}
+
+bool Music::isPlaying(void) const
+{
+  ALenum state;
+
+  alGetSourcei(source, AL_SOURCE_STATE, &state);
+  return state == AL_PLAYING;
+}
+void Music::update(void)
```

Comment:
Indeed, my bad.
It used to be in a while, that's why I was decrementing `processed`, but in this case, `processed` can only be 1 or 0, so an `if` is enough. I just forgot to remove the decrement operation